circle.yml

```diff
@@ -115,5 +157,68 @@ workflows:
   version: 2
   build-test-and-deploy:
     jobs:
-      - build
+      - build:
+          name: quality_migrations_tests
+          tests_group: 0
+          tests_type: clean test.quality test.migrations_missing
+      - build:
+          name: python_unit_tests
+          tests_group: 0
+          tests_type: clean test.unit
+          requires:
+            - quality_migrations_tests
+      - build:
+          name: browser_tests
+          tests_group: 0
+          tests_type: clean test.browser
+          requires:
+            - quality_migrations_tests
+      - build:
+          name: js_tests
+          tests_group: 0
+          tests_type: clean test.js
+          requires:
+            - quality_migrations_tests
+      - build:
+          name: integration_tests_group_1
+          tests_group: 1
+          tests_type: test.integration
+          requires:
+            - python_unit_tests
+            - browser_tests
+            - js_tests
+      - build:
+          name: integration_tests_group_2
+          tests_group: 2
+          tests_type: test.integration
+          requires:
+            - python_unit_tests
+            - browser_tests
+            - js_tests
+      - build:
+          name: integration_tests_group_3
+          tests_group: 3
+          tests_type: test.integration
+          requires:
+            - python_unit_tests
+            - browser_tests
+            - js_tests
+      - coverage:
+          name: coverage
+          requires:
+            - integration_tests_group_1
+            - integration_tests_group_2
+            - integration_tests_group_3
+              #- python_unit_tests
+              #- browser_tests
+              #- js_tests
```

Comment:
In the end we'll need to restrict merging on this job instead of the current build, right?

Reply:
@viadanna What do you mean restrict merging on this job?

Reply:
When a job fails, so we can prevent merging. If this one or a previous one fails, do we need to restrict merging on all jobs or just this one?

Reply:
@viadanna Ah, I see. Well this step only prints out the report it doesn't check for thresholds so it shouldn't fail.
But yes, we probably shouldn't restrict merging on this job since it's purely informational.
The idea is to eventually use this step to send the reports to codecov/coveralls.